### ISSUE TITLE
time zone

### ISSUE THREAD

**Issue by pietercolpaert** [linkedconnections/gtfs2lc#16]
Make sure that timezone is configurable or readable from the feed info

**Comment by pietercolpaert** [comment]
Right now, GTFS2LC always takes the system’s locale. If you want to parse a GTFS file, people should have the right locale timezone set on their system.

**Comment by derhuerst** [comment]
Heads up! [GTFS Time values are *not* relative to midnight, but to noon - 12h](https://gist.github.com/derhuerst/574edc94981a21ef0ce90713f1cff7f6).

[`gtfs2lc` currently just adds the time to midnight](https://github.com/linkedconnections/gtfs2lc/blob/ed01cde4cdbd781104cb9c8944230cfdad70bd8f/lib/ConnectionsBuilder.js#L49-L51). AFAICT, this will fail during DST <-> standard time shifts.

**Comment by derhuerst** [comment]
> Make sure that timezone is configurable or readable from the feed info

Reading it from `stops.stop_timezone` & `agency.agency_timezone` would be the spec-compliant mechanism.

**Comment by pietercolpaert** [comment]
If it were a string concatenation it would have been problematic indeed, but it isn't. We took this into account!

**Comment by derhuerst** [comment]
> If it were a string concatenation it would have been problematic indeed, but it isn't. We took this into account!

Are you sure? Because [the code doesn't look like it](https://github.com/linkedconnections/gtfs2lc/blob/ed01cde4cdbd781104cb9c8944230cfdad70bd8f/lib/ConnectionsBuilder.js#L49-L51).

I couldn't verify that it works because I failed to get `gtfs2lc` running, because of #113 and other issues.

**Comment by pietercolpaert** [comment]
Yes: it adds the duration since midnight to midnight of that date. I remember us bumping in that issue and making sure it’s correctly fixed. Feel free to reopen if you would see unexpected behaviour in your results. @julianrojas87 can you look into #113 so that @derhuerst can continue?

**Comment by derhuerst** [comment]
Just noticed that I had opened #110 for this, so no matter if it's actually an issue, we can discuss there.
